pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,371 @@
+        };
+
+        this.kpatchService = serviceProxy("kpatch");
+
+        this.checkSetup = this.checkSetup.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.onClose = this.onClose.bind(this);
+        this.handleInstall = this.handleInstall.bind(this);
+    }
+
+    // Only current patches or also future ones
+    current(enabled, installed, unavailable) {
+        return enabled && (installed || unavailable);
+    }
+
+    componentDidMount() {
+        check_missing_packages(["kpatch", "kpatch-dnf"])
+                .then(d =>
+                    this.checkSetup().then(() =>
+                        this.setState({
+                            loaded: true,
+                            unavailable: d.unavailable_names || [],
+                            missing: d.missing_names || [],
+                        })
+                    )
+                )
+                .catch(e => console.log("Could not determine kpatch availability:", e));
+
+        this.kpatchService.addEventListener('changed', () => {
+            this.setState(state => {
+                const current = this.current(this.kpatchService.enabled, state.patchInstalled, state.patchUnavailable);
+                return ({
+                    enabled: this.kpatchService.enabled,
+                    justCurrent: current && !state.auto,
+                    applyCheckbox: current,
+                });
+            });
+        });
+    }
+
+    checkSetup() {
+        // TODO - replace both with `dnf kpatch status` once https://github.com/dynup/kpatch-dnf/pull/8 lands
+        const kpatch_promise = cockpit.file("/etc/dnf/plugins/kpatch.conf").read()
+                .then(data => {
+                    if (data) {
+                        const auto = /autoupdate\s*=\s*True/i.test(data);
+                        this.setState((state, _) => {
+                            const current = this.current(state.enabled, state.patchInstalled, state.patchUnavailable);
+                            return ({
+                                auto: !!auto,
+                                justCurrent: current && !auto,
+                                applyCheckbox: current,
+                            });
+                        });
+                    }
+                })
+                .catch(() => true); // Ignore errors, most likely just does not exist
+
+        const uname_promise = cockpit.spawn(["uname", "-r"])
+                .then(data => {
+                    const fields = data.split("-");
+                    const kpp_kernel_version = fields[0].replaceAll(".", "_");
+                    let release = fields[1].split(".");
+                    release = release.slice(0, release.length - 2); // remove el8.x86_64
+                    const kpp_kernel_release = release.join("_");
+                    const patch_name = ["kpatch-patch", kpp_kernel_version, kpp_kernel_release].join("-");
```

Comment:
To confirm, this will also go away in favor of `dnf kpatch status`, right?

Reply:
Indeed, the whole `checkSetup()` should be just call to `dnf kpatch status`.